internal/ledger/compare.go

```diff
@@ -186,24 +186,20 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 
 // diffRecord represents a diverging record in json
 type diffRecord struct {
-	Namespace string          `json:"namespace,omitempty"`
-	Key       string          `json:"key,omitempty"`
-	Record1   *snapshotRecord `json:"snapshotrecord1,omitempty"`
-	Record2   *snapshotRecord `json:"snapshotrecord2,omitempty"`
+	Namespace string      `json:"namespace,omitempty"`
+	Key       string      `json:"key,omitempty"`
+	Record1   interface{} `json:"snapshotrecord1,omitempty"`
+	Record2   interface{} `json:"snapshotrecord2,omitempty"`
```

Comment:
It was better to use the concrete type, as before. On a second thought, as an alternate we could remove "omitempty" in the json annotation for `Record1` and `Record2`. That will force to print a nil object as null in the final JSON -- which I feel would be even clearer than {}.

Sorry for an additional round of change on this.